apollo-integration/src/test/java/com/apollographql/apollo/ApolloWatcherTest.java

```diff
@@ -104,6 +111,50 @@ public void testQueryWatcherUpdated_SameQuery_DifferentResults() throws IOExcept
   }
 
   @Test
+  public void testQueryWatcherUpdated_Store_write() throws IOException, InterruptedException,
+      TimeoutException, ApolloException {
+    final NamedCountDownLatch firstResponseLatch = new NamedCountDownLatch("firstResponseLatch", 1);
+    final NamedCountDownLatch secondResponseLatch = new NamedCountDownLatch("secondResponseLatch", 2);
+
+    EpisodeHeroNameQuery query = EpisodeHeroNameQuery.builder().episode(Episode.EMPIRE).build();
+    server.enqueue(mockResponse("EpisodeHeroNameResponseWithId.json"));
+
+    ApolloQueryWatcher<EpisodeHeroNameQuery.Data> watcher = apolloClient.query(query).watcher();
+    watcher.enqueueAndWatch(
+        new ApolloCall.Callback<EpisodeHeroNameQuery.Data>() {
+          @Override public void onResponse(@Nonnull Response<EpisodeHeroNameQuery.Data> response) {
+            if (secondResponseLatch.getCount() == 2) {
+              assertThat(response.data().hero().name()).isEqualTo("R2-D2");
```

Comment:
Actually, I see RxJava in the dependencies of apollo-integration so maybe we could use that. Is it ok to use `AndroidSchedulers.mainThread()` in there or are we pure java ?